File: solutions/lua/bob/1/bob.lua
local bob = {}

function bob.hey(say)
	if say:match("^%s*$") then
		return "Fine. Be that way!"
	elseif say:match("?$") and say == say:upper() and say:lower() ~= say then
		return "Calm down, I know what I'm doing!"
	elseif say:match("?%s*$") then
		return "Sure."
	elseif say == say:upper() and say:lower() ~= say then
		return "Whoa, chill out!"
	else
		return "Whatever."
	end
end

return bob

File: solutions/python/bob/1/bob.py
import re

def response(hey_bob):
    if re.match(r"^\s*$", hey_bob):
        return "Fine. Be that way!"

    match [hey_bob.strip()[-1] == "?", hey_bob == hey_bob.upper() and bool(re.match(r"[A-Za-z]", hey_bob))]:
        case [True, True]:
            return "Calm down, I know what I'm doing!"
        case [True, _]:
            return "Sure."
        case [_, True]:
            return "Whoa, chill out!"
        case [_, _]:
            return "Whatever."
